src/map/battle.cpp

```diff
@@ -7145,6 +7145,7 @@ static void battle_calc_attack_post_defense(struct Damage* wd, struct block_list
  */
 static void battle_calc_attack_plant(struct Damage* wd, struct block_list *src,struct block_list *target, uint16 skill_id, uint16 skill_lv)
 {
+	std::shared_ptr<s_skill_db> skill = skill_db.find(skill_id);
```

Comment:
Since you only need this variable inside the if further down, you should initialize it in that scope.